Update Kub-recommended resources documentation (#262)
1Password/scim-examples: kubernetes/README.md

# Deploy 1Password SCIM Bridge on Kubernetes

> _Learn how to deploy 1Password SCIM Bridge on a Kubernetes cluster._

Reviewer: These autoformatting changes are actually changing bolded statements to italicized, which we don't want. Can we change back to using `*` ?

Author: Comparing with our [main branch](https://github.com/1Password/scim-examples/tree/master/kubernetes) the text initially represented with *xxx* as italics are now adapted as  _xxx_ by editor. Do we want to make it bold there?

After the DNS record has propagated, set the `OP_TLS_DOMAIN` environment variable to enable the CertificateManager component.

> _Example command:_

Reviewer: Same comment as above

You can also test your SCIM bridge by sending an authenticated SCIM API request.

> _Example command:_

Reviewer: Same comment as above

   ```

   Kubernetes automatically updates the credentials file mounted in the Pod with the new Secret value.

3. Test your SCIM bridge using the new bearer token associated with the regenerated `scimsession` file.
4. Update your identity provider configuration with the new bearer token.

## Appendix: Resource recommendations

> The default resource recommendations for the SCIM bridge and Redis deployments are acceptable in most scenarios, but they may fall short in high-volume deployments where a large number of users and/or groups are being managed. We strongly recommend increasing the resources for both the SCIM bridge and Redis deployments.

Reviewer: Since the performance improvements we've made (see https://docs.google.com/document/d/1cD5GY8kNm-RwqqA1sklwzx77yoknmGV-bfaTEvUv9sg/edit), we no longer need to recommend customers bump resources for their Redis deployments. So this statement can read:

> We strongly recommend increasing the resources for the SCIM bridge deployment.

I would also like to add a comment about deploy resources vs. normal everyday use of the SCIM bridge after this comment. Something like:

> Once all users and groups have been provisioned, it is safe to lower these resources back to their defaults for everyday use of the SCIM bridge.

Author: Thanks. Thats good callout. <br/>
For the second suggestion, I thought we have lines  after the **Default** section but its right to add on top section as well in case one skips to read further.

<details>
  <summary>High Volume Deployment</summary>

```yaml
requests:
  cpu: 500m
  memory: 512M

limits:
  cpu: 1000m
  memory: 1024M
```

</details>

<details>
  <summary>Very High Volume Deployment</summary>

The best practices for Kubernetes resource limits are:

1.  Never set [CPU](https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource) limits to avoid Kubernetes CPU throttling and allow for efficient use of available CPU.
2.  Set [Memory](https://kubernetes.io/docs/tasks/configure-pod-container/assign-memory-resource) limits same as requests to help avoiding OOM kills.

This works well in the deployment scenario where the 1Password SCIM bridge is the only pod in the cluster, so that it has a priority of service and can consume all the resources in the cluster without affecting the performance of other pods.

If you are deploying the SCIM bridge to a cluster with other production services, we recommend using the following resources depending on the size of your deployment.

```yaml
requests:
  cpu: 1000m
  memory: 1024M

limits:

>   memory: 1024M

Reviewer: Is there a reason we left off the CPU limit of `cpu: 2000m`?

Author: So as part of no CPU limit strategy we remove it from limits to in order to decrease occurrences of CPU throttling and make full use of available CPU.

Author: That reads better. Thanks. 😄  
Changes applied.

<details>
  <summary>High Volume Deployment</summary>

```yaml
requests:
  cpu: 500m
  memory: 512M

limits:
  memory: 1024M
```

</details>

<details>
  <summary>Very High Volume Deployment</summary>

The best practices for Kubernetes resource limits are:

1.  Never set [CPU](https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource) limits to avoid Kubernetes CPU throttling and allow for efficient use of available CPU.
2.  Set [Memory](https://kubernetes.io/docs/tasks/configure-pod-container/assign-memory-resource) limits same as requests to help avoiding OOM kills.

This works well in the deployment scenario where the 1Password SCIM bridge is the only pod in the cluster, so that it has a priority of service and can consume all the resources in the cluster without affecting the performance of other pods.

If you are deploying the SCIM bridge to a cluster with other production services, we recommend using the following resources depending on the size of your deployment.

> **Note**
>

> > We are recommending not to set CPU limits when you expect performance. However you can set CPU limits if you have a good usecase to manage resources. It can be modified in your deployment based on our generic example and its expected to work in most circumstances.

Reviewer 2: ```suggestion
> We recommend to not set CPU limits when you expect higher performance. However, you can set CPU limits to manage resources if needed. This can be modified in your deployment based on our generic example, and its expected to work in most circumstances.
```

You can set `OP_REDIS_URL` directly to reboot SCIM bridge and connect to the specified Redis server:

> _Example command:_

Reviewer: Same comment as above
